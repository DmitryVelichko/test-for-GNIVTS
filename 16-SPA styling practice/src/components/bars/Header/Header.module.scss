.bar {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100px;
  padding: 8px;
  background-color: midnightblue;

  .link {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 94px;
    color: #fff;
    font-size: 18px;
    text-decoration: none;
    text-align: center;
    margin: 0 12px;
    height: 30px;

    &:hover {
      color: red;
    }
  }

  .active {
    color: midnightblue;
    background-color: white;
  }
}
