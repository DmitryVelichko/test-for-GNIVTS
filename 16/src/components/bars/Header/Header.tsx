import React from 'react';
import { NavLink } from 'react-router-dom';

import classes from './Header.module.scss';

const Header: React.FC = () => {
  const nav = [
    { title: 'Главная', path: '/' },
    { title: 'О проекте', path: '/about' },
  ];

  return (
    <div className={classes.bar}>
      {nav.map((item) => (
        <NavLink
          activeClassName={classes.active}
          className={classes.link}
          to={item.path}
          key={item.path}
          exact
        >
          {item.title}
        </NavLink>
      ))}
    </div>
  );
};

export default Header;
