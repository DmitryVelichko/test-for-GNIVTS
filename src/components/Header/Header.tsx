import React from 'react';
import {
  Row, Col, Navigation, NavigationAction, Button,
} from '@ff/ui-kit';

import logo from './assets/gnivc-logo.png';

const Header = () => (
  <Row
    style={{ padding: '20px 40px', marginBottom: '30px', backgroundColor: '#224b8d' }}
    justify="space-between"
  >
    <Col xs={12} sm={6}>
      <Row align="middle" colGap="lg" wrap={false}>
        <img
          src={logo}
          alt="gnivc logo"
          style={{ display: 'block', height: '30px' }}
        />
        <Navigation horizontal>
          <NavigationAction active>Каталог</NavigationAction>
          <NavigationAction>Каталог</NavigationAction>
          <NavigationAction>Каталог</NavigationAction>
        </Navigation>
      </Row>
    </Col>

    <Col xs={12} sm={6}>
      <Row wrap={false} colGap="lg" align="middle" justify="end">
        {/* поискать нормальные кнопки */}
        <Button variant="text" type="primary" style={{ fontSize: '16px', color: '#fff' }}>
          Добавить задачу
        </Button>
        <Button variant="text" type="primary" style={{ fontSize: '16px', color: '#fff' }}>
          Имя Фамилия
        </Button>
        <img
          src="#"
          alt="profile logo"
          style={{
            display: 'block',
            width: '30px',
            height: '30px',
            borderRadius: '50%',
            objectFit: 'cover',
          }}
        />
      </Row>
    </Col>
  </Row>
);

export default Header;
