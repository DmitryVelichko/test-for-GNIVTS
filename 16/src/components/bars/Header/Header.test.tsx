import React from 'react';
import { render } from '@testing-library/react';

import Header from '.';

const renderComponent = () => render(<Header />);

describe('Header component', () => {
  it('snapshot matches', () => {
    expect(renderComponent).toMatchSnapshot();
  });
});
