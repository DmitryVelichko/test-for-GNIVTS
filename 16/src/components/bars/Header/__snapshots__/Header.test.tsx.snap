// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot match 1`] = `[Function]`;

exports[`Header component snapshot matches 1`] = `[Function]`;

exports[`Компонент Header snapshot 1`] = `[Function]`;
